### ISSUE TITLE
IndexOutOfBounds exception

### ISSUE THREAD

**Issue by kenbot** [softwaremill/quicklens#35]
The following code:
```
case class Topping(cherries: Int)
case class Cake(toppings: List[Topping])
case class Bakery(cakes: List[Cake])

val myBakery = Bakery(List(Cake(List(Topping(0), Topping(3))), 
                           Cake(List(Topping(4))), 
                           Cake(Nil)))
import com.softwaremill.quicklens._
modify(myBakery)(_.cakes.each.toppings.at(0).cherries).setTo(7)
```
produces the following exception:
```
java.lang.IndexOutOfBoundsException: 0
  at scala.collection.LinearSeqOptimized.apply(LinearSeqOptimized.scala:63)
  at scala.collection.LinearSeqOptimized.apply$(LinearSeqOptimized.scala:61)
  at scala.collection.immutable.List.apply(List.scala:86)
  at com.softwaremill.quicklens.package$$anon$1.at(package.scala:151)
  at .$anonfun$res12$2(<console>:32)
  at $$Lambda$2077/1919422646.apply(Unknown Source)
  at scala.collection.immutable.List.map(List.scala:276)
  at com.softwaremill.quicklens.package$$anon$2.map(package.scala:112)
  at com.softwaremill.quicklens.package$QuicklensFunctor.each(package.scala:101)
  at com.softwaremill.quicklens.package$QuicklensFunctor.each$(package.scala:101)
  at com.softwaremill.quicklens.package$$anon$2.each(package.scala:111)
  at .$anonfun$res12$1(<console>:32)
  at $$Lambda$2075/532145679.apply(Unknown Source)
  at com.softwaremill.quicklens.package$PathModify.setTo(package.scala:69)
  ... 40 elided
```
I would expect it to return the same "Bakery", except with the first Toppings ("0" and "4") set to 7

**Comment by nadavwr** [comment]
This seems to be expected behavior in Quicklens, equivalent to [Monocle's At](https://julien-truffaut.github.io/Monocle/typeclasses/at.html). I wish Quicklens would also support something equivalent to [Monocle's Index](https://julien-truffaut.github.io/Monocle/typeclasses/index.html).

**Comment by adamw** [comment]
@nadavwr sure, that should be quite easy to add. Maybe you'd like to attempt a PR?

**Comment by adamw** [comment]
Released in 1.6.0
